src/Chemistry.hpp

```diff
@@ -142,6 +152,19 @@ template <typename problem_t> void computeChemistry(amrex::MultiFab &mf, const R
 				state(i, j, k, HydroSystem<problem_t>::scalar0_index + nn) = inmfracs[nn] * rho; // scale by rho to return partial densities
 			}
 		});
+
+#if defined(AMREX_USE_HIP)
+		amrex::Gpu::streamSynchronize(); // otherwise HIP may fail to allocate the necessary resources.
```

Comment:
@BenWibking why should this be defined only if HIP is used, instead of being defined for all GPU? As in, shouldn't the correct ifdef be `if defined(AMREX_USE_GPU)`?